Jack branch

JackLaFreniere opened the pull request:
Added changes to detect April Tags better. Adjusted Turret code for slightly different positions and PIDs for the Refined Turret Command. Added code for the Endgame climbing.

JackLaFreniere committed:
2/18/24 6:57
Added more logging for the Limelights and Swerve Drive

jackrschumacher committed:
Merge remote-tracking branch 'origin/master' into Jack-Branch

Evan-Callies committed:
7:23 pm 2/20/24
Evan C.
Created endgame implementation.

Evan-Callies committed:
7:31 pm 2/20/24
Evan
Changed the subsystem used for Endgame

Evan-Callies committed:
7:40 pm 2/20/24
Evan
-Made a setPull Method in TalonPosIO
-Implemented the Method in Elevator Subsystem

JackLaFreniere committed:
Merge remote-tracking branch 'origin/master' into Jack-Branch

JackLaFreniere committed:
2/21/24 5:36
Fixed erroring code

JackLaFreniere committed:
2/21/24 6:00
Made the TurretAimCommand track 0.5 meters behind the speaker April Tags

JackLaFreniere committed:
2/21/24 6:38
Refactored the setPull method to the TurretRefineCommand and adjusted it so now we can have custom MM values.

jackrschumacher committed:
2/21/24 6:56pm
Evan
Re-organized constants for the new elevator DynamicMotionMagic

jackrschumacher committed:
Merge branch 'Jack-Branch' of https://github.com/FRC930/Robot2024 into Jack-Branch

JackLaFreniere committed:
Merge branch 'Jack-Branch' of https://github.com/FRC930/Robot2024 into Jack-Branch

JackLaFreniere committed:
2/21/24 8:21
Made a change in the TurretRefineCommand to slightly adjust values when we are off-centered and close to the speaker.

JackLaFreniere committed:
2/22/24 7:12
Adjusted Limelight code to update Odometry better from farther away

JackLaFreniere committed:
2/22/24 7:17
Added the Limelight configuration files into the Configuration Files folder

jackrschumacher commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -237,6 +240,26 @@ public class RobotContainer {
+    public static final boolean TURRET_REFINE_COMMAND_LIMIT_FORWARD_MOTION = false;
+    public static final boolean TURRET_REFINE_COMMAND_LIMIT_REVERSE_MOTION = false;
+
+    public static final boolean ENDGAME_ELEVATOR_ENABLEFOC = false;
+    public static final double ENDGAME_ELEVATOR_JERK = 0;
+    public static final double ENDGAME_ELEVATOR_ACCELERATION = 0;
+    public static final double ENDGAME_ELEVATOR_VELOCITY = 0;
+    public static final boolean ENDGAME_ELEVATOR_OVERRIDEBRAKEDURNEUTRAL = false;
+    public static final int ENDGAME_ELEVATOR_SLOT = 0;
+    public static final double ENDGAME_ELEVATOR_FEEDFORWARD = 0;
+    public static final boolean ENDGAME_ELEVATOR_LIMITFORWARDMOTION = false;
+    public static final boolean ENDGAME_ELEVATOR_LIMITREVERSEMOTION = false;
```
Copy PIDs from elevator @JackLaFreniere

JackLaFreniere committed:
2/22/24 7:32
Copied over PID values for the shooter elevator for endgame climb.

JackLaFreniere committed:
2/22/24 7:36
Re-implemented the booleans to enable or disable updating april tags and updating odometry

JackLaFreniere committed:
2/22/24 7:45
Added comments to some of the Limelight code methods

JackLaFreniere committed:
2/22/24 7:46
Made the command to use the actual values of the April Tags and made constants for the sorting values.

jackrschumacher reviewed (approved):
Reviewed by Jack S, Steve, Jack L and Matt
